[WebDriver] Enable webdriver-bidi flag by default in run-webdriver-tests

lauromoura opened the pull request:
#### 7dca8b8440a029d1250b91925e49df7cd6e5dc0c
<pre>
[WebDriver] Enable webdriver-bidi flag by default in run-webdriver-tests
<a href="https://bugs.webkit.org/show_bug.cgi?id=286902">https://bugs.webkit.org/show_bug.cgi?id=286902</a>

Reviewed by Carlos Alberto Lopez Perez.

Add an option `--disable-webdriver-bidi`, making it opt-out, as the
GTK/WPE bots are the only ones in the post-commit buildbot using it.

* Tools/Scripts/run-webdriver-tests:

Canonical link: <a href="https://commits.webkit.org/290436@main">https://commits.webkit.org/290436@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/617d829d77b141274d5976d7598ed3afba9301de

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/90049 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/9578 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/44950 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/95049 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/40822 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/92101 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/9965 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/17868 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/69319 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/26921 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/93050 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/132/builds/7625 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/81686 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/49681 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/7353 "Passed tests") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/64/builds/36043 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/39956 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/77685 "Passed tests") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/63/builds/37113 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/96875 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/17237 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/12644 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/78319 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/17493 "Built successfully") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/8/builds/77510 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/77525 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/19137 "Built successfully and passed tests") | [  ~~🧪 vision-wk2~~](https://ews-build.webkit.org/#/builders/126/builds/21980 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-intel-wk2~~](https://ews-build.webkit.org/#/builders/119/builds/20563 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/10431 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/17247 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/16988 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/20440 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/18771 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

clopez commented on Tools/CISupport/build-webkit-org/config.json:
```diff
@@ -690,11 +692,13 @@
                   {
                     "name": "WPE-Linux-64-bit-Debug-WebDriver-Tests", "factory": "TestWebDriverFactory",
                     "platform": "wpe", "configuration": "debug", "architectures": ["x86_64"],
+                    "additionalArguments": ["--enable-webdriver-bidi"],
```
It looks we are enabling WebDriver BiDi on all the bots.

So I wonder if instead of passing the flag here to all the bots, maybe it makes more sense to switch the default in the script `run-webdriver-tests` so it runs with bidi enabled by default (unless `--disable-webdriver-bidi` is passed to it)?

That way we don't need to pass any extra flag here.

lauromoura replied:
Good catch. I updated the PR making it opt-out of `run-webdriver-tests`.

clopez commented on Tools/Scripts/run-webdriver-tests:
```diff
@@ -53,8 +53,12 @@ option_parser.add_option('--display-server', choices=['xvfb', 'xorg', 'weston',
+                         help='Run Selenium tests with BiDi enabled')
+option_parser.add_option('--disable-webdriver-bidi', action='store_const',
+                         const=False, dest='enable_webdriver_bidi',
+                         help='Run Selenium tests with BiDi disabled')
```
The `--enable-webdriver-bidi` switch looks a bit redundant and can be confusing regarding to what one is the current default. I would just remove it and keep only the `--disable-webdriver-bidi` one.

But is fine in any case.

lauromoura committed:
[WebDriver] Enable webdriver-bidi flag by default in run-webdriver-tests
https://bugs.webkit.org/show_bug.cgi?id=286902

Reviewed by Carlos Alberto Lopez Perez.

Add an option `--disable-webdriver-bidi`, making it opt-out, as the
GTK/WPE bots are the only ones in the post-commit buildbot using it.

* Tools/Scripts/run-webdriver-tests:

Canonical link: https://commits.webkit.org/290436@main
